packages/headless/src/controllers/core/search-parameter-manager/headless-core-search-parameter-manager.ts

```diff
@@ -201,10 +202,30 @@ function getFacets(state: Partial<SearchParametersState>) {
   return Object.keys(f).length ? {f} : {};
 }
 
+function getExcludedFacets(state: Partial<SearchParametersState>) {
```

Comment:
This seems a near exact copy paste of `getSelectFacets`, but with the mapping function doing a different operation.

This could be refactored to use one function for both

Suggestion: 

```typescript
function getFacets(
  state: Partial<SearchParametersState>,
  valuesSelector: (currentValues: FacetValueRequest[]) => string[],
  out: keyof SearchParameters
) {
  if (state.facetSet === undefined) {
    return {};
  }

  const facets = Object.entries(state.facetSet)
    .filter(([facetId]) => state.facetOptions?.facets[facetId]?.enabled ?? true)
    .map(([facetId, {request}]) => {
      const facetValues = valuesSelector(request.currentValues);
      return facetValues.length ? {[facetId]: facetValues} : {};
    })
    .reduce((acc, obj) => ({...acc, ...obj}), {});

  return Object.keys(facets).length ? {[out]: facets} : {};
}
```

and then on line 132-133:

```typescript
...getFacets(state, getSelectedValues, 'f'),
...getFacets(state, getExcludedValues, 'fExcluded'),
```

Reply:
Good idea, I'll apply right now 🙂